### PR TITLE
bsunanda:Phase2-hgx59 Add beam line for HGCal test beam studies

### PR DESCRIPTION
**Pull request by bsunanda**
Possibility of readout of the trigger scintillators

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HcalCommonData/src/HcalGeomParameters.cc [hunk 1]
```diff
@@ -15,7 +15,7 @@
 #include "CLHEP/Units/GlobalPhysicalConstants.h"
 #include "CLHEP/Units/GlobalSystemOfUnits.h"
 
-//#define DebugLog
+#define DebugLog
```
Comment (ianna) [inline Geometry/HcalCommonData/src/HcalGeomParameters.cc]: @bsunanda - please, comment it out for release or use LogDebug

File: SimG4Core/Geometry/src/DDG4Builder.cc [hunk 2]
```diff
@@ -41,7 +41,7 @@ DDG4Builder::~DDG4Builder() {
 }
 
 G4LogicalVolume * DDG4Builder::convertLV(const DDLogicalPart & part) {
-  LogDebug("SimG4CoreGeometry") << "DDG4Builder::convertLV(): DDLogicalPart = " << part;
+  std::cout << "DDG4Builder::convertLV(): DDLogicalPart = " << part << "\n";
```
Comment (ianna) [inline SimG4Core/Geometry/src/DDG4Builder.cc]: @bsunanda - please, change it back to LogDebug for release

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HcalCommonData/src/HcalGeomParameters.cc (hunk 1 above)
Comment (ianna) [inline Geometry/HcalCommonData/src/HcalGeomParameters.cc]: @bsunanda - please, comment it out.

**Comment by bsunanda** [comment]
@ianna @civanch Please approve

**Comment by ianna** [comment]
@bsunanda - there is a debug statements printed due to #define Debug (see my comment above). Please, comment it out.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by ianna** [comment]
+1

**Comment by bsunanda** [comment]
@civanch Could you sign this please

**Comment by civanch** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Comment by Dr15Jones** [comment]
This seems to have caused a thread-safety problem found by the static analysis tool
```
In call stack '  OscarMTProducer::produce() calls function  RunManagerMTWorker::produce() calls function  RunManagerMTWorker::initializeThread() calls function  AttachSD::create() const calls function  SensitiveDetectorMakerBase::make() const virtual calls function  SensitiveDetectorMaker<HGCalTB16SD01>::make() const calls function  HGCalTB16SD01::HGCalTB16SD01() calls function  DDName::name() const calls function  DDI::Singleton<std::vector<std::_Rb_tree_const_iterator<std::pair<const std::pair<std::basic_string<char>, std::basic_string<char> >, int> >, std::allocator<std::_Rb_tree_const_iterator<std::pair<const std::pair<std::basic_string<char>, std::basic_string<char> >, int> > > > >::instance() static variable  DDI::Singleton<std::vector<std::_Rb_tree_const_iterator<std::pair<const std::pair<std::basic_string<char>, std::basic_string<char> >, int> >, std::allocator<std::_Rb_tree_const_iterator<std::pair<const std::pair<std::basic_string<char>, std::basic_string<char> >, int> > > > >::instance()::value' is accessed , 'OscarMTProducer::produce()' overrides 'edm::stream::EDProducerBase::produce() virtual'
```

@ianna can you comment on if this is really a problem?

**Comment by Dr15Jones** [comment]
A quick look at some of the other sensitive detectors didn't turn up any cases where the `DDName` is ever called.

File: SimG4CMS/HGCalTestBeam/src/HGCalTB16SD01.cc [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+  DDValue           ddv(attribute,name,0);
+  filter.setCriteria(ddv,DDCompOp::equals);
+  DDFilteredView fv(cpv);
+  fv.addFilter(filter);
+  bool dodet = fv.firstChild();
+
+  std::map<G4String,int> matNames;
+  while (dodet) {
+    const DDLogicalPart & log = fv.logicalPart();
+    matName = log.material().name().name();
```
Comment (Dr15Jones) [inline SimG4CMS/HGCalTestBeam/src/HGCalTB16SD01.cc]: I think this is the thread-unsafe operation.

**Pull request merged**